src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -912,32 +910,47 @@ public void replaceVariablesInFileRecursive(String startPath, Map<String, String
      *
      * @param filePath     the path where the file is located
      * @param replacements the replacements that should be applied
-     * @throws IOException if an issue occurs on file access for the replacement of the variables.
      */
-    public void replaceVariablesInFile(String filePath, Map<String, String> replacements) throws IOException {
+    public void replaceVariablesInFile(Path filePath, Map<String, String> replacements) {
         log.debug("Replacing {} in file {}", replacements, filePath);
         // https://stackoverflow.com/questions/3935791/find-and-replace-words-lines-in-a-file
-        Path replaceFilePath = Path.of(filePath);
-        Charset charset = StandardCharsets.UTF_8;
 
-        String fileContent = Files.readString(replaceFilePath, charset);
-        for (Map.Entry<String, String> replacement : replacements.entrySet()) {
-            fileContent = fileContent.replace(replacement.getKey(), replacement.getValue());
+        try {
+            if (isBinaryFile(filePath)) {
+                // do not try to read binary files with 'readString'
+                return;
+            }
+            // Note: we cannot check if this is a binary file or not. We try to read it, in case it fails, we only log this below, but continue
```

Comment:
I don't understand this note. If we cannot check this, why do we do something that looks like a check?
Looking at the code, I'd say the note is wrong. Can you explain this? Maybe rephrase the note?

Reply:
I initially wanted to have a universal checks, but this seems to be very difficult like mentioned here: https://stackoverflow.com/questions/620993/determining-binary-text-file-type-in-java

So there is no 100% guarantee and we need to run into the exception.

Then I thought, we can at least implement a simply check for typical binary files such images, however this will never be exhaustive and only prevents the exception from happening too often.

Reply:
Thanks for the explanation. The updated note explains it very well.